pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/kubernetes/KubernetesRuntime.java

```diff
@@ -855,13 +854,8 @@ protected List<String> getExecutorCommand() {
     }
 
     private List<String> getDownloadCommand(Function.FunctionDetails functionDetails, String userCodeFilePath) {
-        if (Arrays.stream(PackageType.values()).anyMatch(type ->
```

Comment:
Actually, this download method is similar to line 867. The only difference is we save the function in a different place. We provide the [package management](https://docs.google.com/document/d/1FDSNhc8YB1yUMFBUrx8p-1nH1s5Kftp04PkFKWec-2U/edit) to save the functions. Should we fix the packageUrl at the register side not remove this?

Reply:
Since we can download anyway the function itself directly, is there value to do it ?
Can this be discussed and done in another PR ?

Reply:
Agreed with @zymap, the real issue here is the `packageUrl` is not been set properly, but not the "unused" issue. The correct way is to make the `packageUrl` set when the user creates functions with package URLs from the package management service.

Reply:
What's the advantage compared to downloading the function ?

Reply:
@cbornet the package management service has versioning on packages, and it can support different storage backend other than bookkeeper DLog.

Reply:
IIRC, the `functions download` will go for the files stored in Bookkeeper DLog only, and will not download packages from the package management service. So if the user provides `packageUrl`, it should use `pulsar-admin packages download`.

Reply:
No `functions download` will [download the package](https://github.com/apache/pulsar/blob/172a84624f47bb722e6b419de4e239fbb8ecc154/pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java#L1483) when appropriate.

Reply:
I think that "functions download" should work as expected, that is to download from DLog or from PackageManagement

do you have any pointers about the problem ?

Reply:
@freeznet @zyma considering that `function download` downloads the package and not from BK when needed, do you still have issue with this PR ?

Reply:
IIRC, when you run this in Kubernetes, the pod needs to download the function from the function worker. This [place](https://github.com/apache/pulsar/blob/172a84624f47bb722e6b419de4e239fbb8ecc154/pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java#L1481) only download the function in the function worker and do some validates. So the download command will run in the function instance and then start the instance. 
So my intention is we should add the packageUrl when the function is uploading by a package management URL. So the Kubernetes runtime can get the function package from the package management service, not from the function worker.

Reply:
Sorry to be insistent but this code [downloads the package](https://github.com/apache/pulsar/blob/172a84624f47bb722e6b419de4e239fbb8ecc154/pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/FunctionsImpl.java#L852) using the admin client as the `downloadPackageFile` name of the function suggests. It would be a bug if it were not so.